### PR TITLE
[Doc] Updating instructions for adding dependencies

### PR DESCRIPTION
**Pull request by ppadala**
Instructions for adding a new dependency

### CODE REVIEW

**Comment by nathanleclaire** [comment]
Very nice, thanks for it @ppadala !

LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -148,11 +148,26 @@ Scrub build results:
     make lint
     make dco
 
-### Save and restore dependencies
+### Restore, update and save dependencies
+
+When you make a fresh copy of the repo, all the dependecies are in `vendor/` directory for the builds to work. If you want to update the dependencies
```
Comment (nathanleclaire) [inline CONTRIBUTING.md]: Oof, didn't catch this before, but it seems to be a sentence fragment.

Comment (ppadala) [inline CONTRIBUTING.md]: I think it reads fine in the rendered page. But, let me know if you want me to change something specific.

Comment (nathanleclaire) [inline CONTRIBUTING.md]: It's mostly an internal document so I think we can let it slide.

**Pull request merged**